docs/user-guide/docker/building-in-docker.rst

```diff
@@ -0,0 +1,244 @@
+    Docker container. You can do this by either mounting the file or by including it in your
+    own custom image (e.g. if the custom component or policy has extra dependencies). Make sure
+    that your module is in the Python module search path by setting the
+    environment variable ``PYTHONPATH=$PYTHONPATH:<directory of your module>``.
+
+Adding Custom Actions
+*********************
+
+To create more sophisticated assistants, you will want to use :ref:`custom-actions`.
+Continuing the example from above, you might want to add an action which tells
+the user a joke to cheer them up.
+
+Running a Custom Action in Docker-Compose
+#########################################
+
+Start by creating the custom actions in a directory ``actions``:
+
+.. code-block:: bash
+
+  mkdir actions
+  # Rasa SDK expects a python module.
+  # Therefore, make sure that you have this file in the directory.
+  touch actions/__init__.py
+  touch actions/actions.py
+
+Then build a custom action using the Rasa SDK, e.g.:
+
+.. code-block:: python
+
+  import requests
+  import json
+  from rasa_sdk import Action
+
+
+  class ActionJoke(Action):
+    def name(self):
+      return "action_joke"
+
+    def run(self, dispatcher, tracker, domain):
+      request = requests.get('http://api.icndb.com/jokes/random').json()  # make an api call
+      joke = request['value']['joke']  # extract a joke from returned json response
+      dispatcher.utter_message(text=joke)  # send the message back to the user
+      return []
+
+Next, add the custom action in your stories and your domain file.
+Continuing with the example bot from ``rasa init``, replace ``utter_cheer_up`` in
+``data/stories.md`` with the custom action ``action_joke``, and add
+``action_joke`` to the actions in the domain file.
+
+To instruct Rasa to use the action server you have to tell Rasa its location.
+Add this to your ``endpoints.yml`` (if it does not exist, create it):
+
+.. code-block:: yaml
+
+  action_endpoint:
+    url: http://app:5055/webhook
+
+To spin up the action server together with the Rasa instance, add a service
+``app`` to the ``docker-compose.yml``:
+
+.. code-block:: yaml
+   :emphasize-lines: 11-14
+
+   version: '3.0'
+   services:
+     rasa:
+       image: rasa/rasa:latest-full
+       ports:
+         - 5005:5005
+       volumes:
+         - ./:/app
+       command:
+         - run
+     app:
+       image: rasa/rasa-sdk:latest
+       volumes:
+         - ./actions:/app/actions
+
+This pulls the image for the Rasa SDK which includes the action server,
+mounts your custom actions into it, and starts the server.
+
+Run ``docker-compose up`` to start the action server together with Rasa.
+
+.. warning::
+
+   If you create a more complicated
+   action that has extra library dependencies, you will need to
+   :ref:`build an action server image<building-an-action-server-image>` to run your code.
+
+Deploying your Model
+####################
+
+Once you're happy with your model, you can
```

Comment:
what happened here

Reply:
Hahaha, i was hoping you'd look at the commit before the split bc I haven't figured out exactly where the pages split yet. This will link to the deployment page lol